### PR TITLE
The dot-membrane -- synchronous passable-only revocable membrane

### PR DESCRIPTION
**Pull request by erights**

### CODE REVIEW

File: packages/marshal/src/dot-membrane.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      }
+      default: {
+        assert.fail(X`unrecognized passStyle ${passStyle}`);
+      }
+    }
+    mineToYours.set(mine, yours);
+    yoursToMine.set(yours, mine);
+    return yours;
+  };
+  // We need to pass this while convertYoursToMine is still in termporal
```
Comment (michaelfig) [inline packages/marshal/src/dot-membrane.js]:
```suggestion
  // We need to pass this while convertYoursToMine is still in temporal
```

**Review by michaelfig — COMMENTED** [review]
LGSF.  Looks good so far!

Please request another review from me when you actually want one.

**Comment by erights** [comment]
> Please request another review from me when you actually want one.
@michaelfig , I want one, thanks!

@katelynsills @dtribble, I think this should be adequate to start on the use-object pattern we talked about. No JS Proxies!

This should do logically the right thing. I'm only using marshal as a shortcut. Since both sides are local, this can be optimized to something much cheaper.

File: packages/marshal/src/dot-membrane.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            yourResolve(pass(myFulfillment));
+          },
+          myReason => {
+            yourReject(pass(myReason));
+          },
+        ).catch(metaReason => {
+          // This can happen if myFulfillment or myReason is not passable.
+          // TODO verify that metaReason must be your-side-safe
+          yourReject(metaReason);
+        });
```
Comment (michaelfig) [inline packages/marshal/src/dot-membrane.js]: I'd like to give a best-effort attempt to pass the metaReason, and handle when the catch block could still throw.

```suggestion
          yourReject(pass(metaReason));
        }).catch(metaMetaReason => yourReject(metaMetaReason));
```

**Review by michaelfig — APPROVED** [review]
Looks good!  I'm happy to see this code actually land after we had sketched it out so long ago. :)

I'd like to try closing the door on a `metaError` leak, but I'm not sure my suggestion is the best way forward.  YMMV.

File: packages/marshal/src/dot-membrane.js (hunk 2 above)
Comment (erights) [reply packages/marshal/src/dot-membrane.js]: It is hard to tell what the right thing is here, but I like your's better than mine. Done. Resolving.

**Pull request merged**